### PR TITLE
Remove old Bierman restconf

### PR DESCRIPTION
**Pull request by Tobianas**

### CODE REVIEW

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 1]
```diff
@@ -10,14 +10,11 @@
 import com.google.common.base.Stopwatch;
 import com.google.common.base.Throwables;
 import io.lighty.core.controller.api.AbstractLightyModule;
-import io.lighty.modules.northbound.restconf.community.impl.config.JsonRestConfServiceType;
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:13]: When you look into `import io.lighty.modules.northbound.restconf.community.impl.config.JsonRestConfServiceType;
` you can see that `DRAFT_02` is still used.

Please remove `JsonRestConfServiceType` completely.

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 2]
```diff
@@ -94,59 +78,34 @@ public CommunityRestConf(final DOMDataBroker domDataBroker, final DOMSchemaServi
...
-                            this.domNotificationService, this.domSchemaService, streamsConfiguration);
-                final ServletContainer servletContainer8040 = new ServletContainer(ResourceConfig
-                    .forApplication(restconfApplication8040));
-                jaxrs = new ServletHolder(servletContainer8040);
-                break;
-            default:
-                throw new UnsupportedOperationException("unsupported restconf service type: "
-                    + this.jsonRestconfServiceType.name());
-        }
+        RestconfApplication restconfApplication = new RestconfApplication(schemaCtxHandler,
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:102]: `final`

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 3]
```diff
@@ -94,59 +78,34 @@ public CommunityRestConf(final DOMDataBroker domDataBroker, final DOMSchemaServi
...
+                this.domMountPointService, this.domDataBroker, this.domRpcService, this.domActionService,
+                this.domNotificationService, this.domSchemaService, streamsConfiguration);
+        final ServletContainer servletContainer8040 = new ServletContainer(ResourceConfig
+                .forApplication(restconfApplication));
+        ServletHolder jaxrs = new ServletHolder(servletContainer8040);
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:107]: `final` - That's our code-style.

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 4]
```diff
@@ -195,10 +148,6 @@ protected boolean stopProcedure() {
             this.schemaCtxHandler.close();
             LOG.info("Schema context handler closed");
         }
-        if (this.restconfProvider != null) {
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:198]: Why have you removed this?

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 5]
```diff
@@ -229,16 +178,4 @@ public void startServer() {
         LOG.info("Jetty started");
     }
 
-    public RestconfImpl getRestconfImpl() {
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:232]: Are they deprecated - why these are removed?

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConfBuilder.java [hunk 6]
```diff
@@ -34,15 +32,6 @@ public static CommunityRestConfBuilder from(final RestConfConfiguration configur
         return new CommunityRestConfBuilder(configuration);
     }
 
-    /**
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConfBuilder.java:37]: Why removed?

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 7]
```diff
@@ -54,34 +45,27 @@ public class CommunityRestConf extends AbstractLightyModule {
...
         this.domRpcService = domRpcService;
         this.domActionService = domActionService;
         this.domNotificationService = domNotificationService;
         this.domMountPointService = domMountPointService;
         this.lightyServerBuilder = serverBuilder;
         this.webSocketPort = new PortNumber(Uint16.valueOf(webSocketPort));
-        this.jsonRestconfServiceType = jsonRestconfServiceType;
         this.domSchemaService = domSchemaService;
         this.httpPort = httpPort;
         this.inetAddress = inetAddress;
```
Comment (PeterSuna) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:71]: Please update documentation down here.

     /* In standard Oxygen ODL Restconf blueprints there are exposed two additional services:
           - jsonRestconfServiceDraft02 (for bierman implementation)
           - jsonRestconfServiceDraft02 (for bierman implementation)
           - jsonRestconfService (for RFC8040 implementation)
           - jsonRestconfService (for RFC8040 implementation)
           Those services are not expected to be required thus they are not here.
           Those services are not expected to be required thus they are not here.
         */
         
Also, search in ligty for "Bierman" or "Draft02" and update documentation.

**Review by PeterSuna — CHANGES_REQUESTED** [review]
Please also rework Swagger module. There is initializing swagger for Bierman "Draft02" Restconf.

https://github.com/PANTHEONtech/lighty/blob/master/lighty-modules/lighty-swagger/src/main/java/io/lighty/swagger/SwaggerLighty.java#L46

**Review by PeterSuna — CHANGES_REQUESTED** [review]
Please drop this commit [Remove bierman support from swagger](https://github.com/PANTHEONtech/lighty/pull/1016/commits/0f859c0c2cd3229b7196e6daea58d4b9f30118ad)

Tests should be removed together with testing code. And this:
https://github.com/PANTHEONtech/lighty/pull/1016/commits/0f859c0c2cd3229b7196e6daea58d4b9f30118ad#diff-fabde05b02e5ef738875ff318676cc8ab39a2c6c9f8119faf6ee24ec55affccdL59
Could be moved to `Remove bierman restconf` commit.

**Review by PeterSuna — APPROVED** [review]

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java [hunk 8]
```diff
@@ -67,13 +57,13 @@ public class CommunityRestConf extends AbstractLightyModule {
...
 
-    public CommunityRestConf(final DOMDataBroker domDataBroker, final DOMSchemaService schemaService,
-            final DOMRpcService domRpcService, final DOMActionService domActionService,
-            final DOMNotificationService domNotificationService, final DOMMountPointService domMountPointService,
-            final int webSocketPort, final JsonRestConfServiceType jsonRestconfServiceType,
-            final DOMSchemaService domSchemaService, final InetAddress inetAddress, final int httpPort,
-            final String restconfServletContextPath, final ExecutorService executorService,
-            final LightyServerBuilder serverBuilder) {
+    public CommunityRestConf(final DOMDataBroker domDataBroker, final DOMRpcService domRpcService,
+             final DOMActionService domActionService, final DOMNotificationService domNotificationService,
```
Comment (ihrasko) [inline lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:54]: Please fix formatting.

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConfBuilder.java (hunk 6 above)
Comment (ihrasko) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConfBuilder.java:37]: OK

File: lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app-helm/helm/lighty-rcgnmi-app-helm/templates/configmaps.yaml [hunk 9]
```diff
@@ -54,9 +54,7 @@ data:
       "restconf":{
         "inetAddress":"0.0.0.0",
         "httpPort":8888,
-        "webSocketPort": 8181,
```
Comment (ihrasko) [inline lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app-helm/helm/lighty-rcgnmi-app-helm/templates/configmaps.yaml:57]: Are you sure that there is no websocket in RFC8040?

File: lighty-modules/lighty-swagger/src/test/java/io/lighty/swagger/SwaggerLightyDRAFT18Test.java [hunk 10]
```diff
@@ -17,9 +16,6 @@ public class SwaggerLightyDRAFT18Test extends SwaggerLightyTest {
     private static final String SWAGGER2_BASE_URI = "http://localhost:8888/apidoc/swagger2/18/apis";
     private static final String OPENAPI3_BASE_URI = "http://localhost:8888/apidoc/openapi3/18/apis";
 
-    protected SwaggerLightyDRAFT18Test() {
```
Comment (ihrasko) [inline lighty-modules/lighty-swagger/src/test/java/io/lighty/swagger/SwaggerLightyDRAFT18Test.java:20]: Please rename files called `DRAFT18` to `RFC8040`.

File: lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app/src/main/resources/example-config/example_config.json [hunk 11]
```diff
@@ -46,8 +46,6 @@
     "restconf":{
         "inetAddress":"0.0.0.0",
         "httpPort":8888,
-        "webSocketPort": 8181,
-        "restconfServletContextPath":"/restconf",
```
Comment (ihrasko) [inline lighty-applications/lighty-rcgnmi-app-aggregator/lighty-rcgnmi-app/src/main/resources/example-config/example_config.json:50]: IMO: there is no need to remove `restconfServletContextPath` configuration.

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java (hunk 1 above)
Comment (ihrasko) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:13]: OK

File: lighty-examples/lighty-community-restconf-actions-app/README.md [hunk 12]
```diff
@@ -88,8 +88,7 @@ No data are modified during the execution. The implementation is using binding A
 ##### Swagger UI
 This application example has active [Swagger](https://swagger.io/) UI for RESTCONF.
 
-URLs for Swagger (choose RESTCONF [draft18](https://tools.ietf.org/html/draft-ietf-netconf-restconf-18) or
-[draft02](https://tools.ietf.org/html/draft-bierman-netconf-restconf-02) in upper right corner):
+URLs for Swagger (choose RESTCONF [draft18](https://tools.ietf.org/html/draft-ietf-netconf-restconf-18) in upper right corner):
```
Comment (ihrasko) [inline lighty-examples/lighty-community-restconf-actions-app/README.md:91]: There is nothing to choose from :D

File: lighty-examples/lighty-community-restconf-netconf-app/README.md [hunk 13]
```diff
@@ -37,8 +37,7 @@ RESTCONF web interface is available at URL ```http://localhost:8888/restconf/*``
 ##### Swagger UI
 This application example has active [Swagger](https://swagger.io/) UI for RESTCONF.
 
-URLs for Swagger (choose RESTCONF [draft18](https://tools.ietf.org/html/draft-ietf-netconf-restconf-18) or
-[draft02](https://tools.ietf.org/html/draft-bierman-netconf-restconf-02) in upper right corner):
+URLs for Swagger (choose RESTCONF [draft18](https://tools.ietf.org/html/draft-ietf-netconf-restconf-18) in upper right corner):
```
Comment (ihrasko) [inline lighty-examples/lighty-community-restconf-netconf-app/README.md:40]: No more choices :D

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java (hunk 8 above)
Comment (ihrasko) [reply lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/CommunityRestConf.java:54]: You are using indent of 5 spaces - and that's not correct.

File: lighty-modules/lighty-swagger/src/test/java/io/lighty/swagger/SwaggerLightyDRAFT18Test.java (hunk 10 above)
Comment (ihrasko) [reply lighty-modules/lighty-swagger/src/test/java/io/lighty/swagger/SwaggerLightyDRAFT18Test.java:20]: In a separate commit you can rename draft18 classes to rfc8040.

**Review by ihrasko — CHANGES_REQUESTED** [review]

**Review by ihrasko — APPROVED** [review]

**Pull request merged**